src/store/store.js

```diff
@@ -2,82 +2,112 @@ import {makeAutoObservable} from 'mobx';
 
 class Store {
   BASE_URL = 'http://192.168.100.106:1339';
+
   categories = [];
+
   currentCategory = [];
+
   questionSets;
+
   chosenQuestionsSet = [];
+
   questions;
+
   currentQuestion = [];
+
   currentAnswers = [];
+
   chosenMode = '';
+
   showHelpDialog = false;
+
   isChooseModeDialog = false;
 
+  isAnswerHasChosen = false;
+
   constructor() {
-    (async () => {
-      await this.fetchCategories();
-      await this.fetchQuestionSets();
-      await this.fetchQuestions();
-    })();
+    this.fetchCategories().catch((error) => console.log('fetchCategories error ', error));
+    this.fetchQuestionSets().catch((error) => console.log('fetchQuestionSets error ', error));
+    this.fetchQuestions().catch((error) => console.log('fetchQuestions error', error));
     makeAutoObservable(this);
```

Comment:
hmmmm, what if your observable variables will become observables after everything is fetched? I mean, you fetch them first and only after that you call makeAutoObservable. Maybe it makes sense to call makeAutoObservable first